common/lib/dependabot/git_commit_checker.rb

```diff
@@ -52,17 +52,17 @@ def pinned?
       # If the specified `ref` is actually a tag, we're pinned
       return true if local_upload_pack.match?(%r{ refs/tags/#{ref}$})
 
-      # If the specified `ref` is actually a branch, we're NOT pinned
-      return false if local_upload_pack.match?(%r{ refs/heads/#{ref}$})
+      # Assume we're pinned unless the specified `ref` is actually a branch
+      return true unless local_upload_pack.match?(%r{ refs/heads/#{ref}$})
 
-      # Otherwise, assume we're pinned
-      true
+      # If the specified `ref` is actually a branch, we're pinned if the branch looks like a version
+      version_tag?(ref)
```

Comment:
❓ Could there be any unintended consequences of making this change in the `common/lib`?

The release branch pattern is very common to `github-actions` and this makes total sense for that ecosystem, but I'm a little concerned that this change in behaviour may mean that in other languages where people are tracking a version branch for other reasons we will start suggesting upgrades where we haven't before 🤔

Reply:
🤔 good point!

Seems a very edge case in other ecosystems, and I'm not sure what user's expectation is, but probably they don't want an update? Not really sure...

I made a similar change about a month ago: https://github.com/dependabot/dependabot-core/commit/e1a8c49f24e41b4ad368aff3a7d35b76aa8c96cb, by the way 😬.

What should we do? Should I override `pinned?` with this logic just for Actions?

Reply:
I believe https://github.com/dependabot/dependabot-core/commit/e1a8c49f24e41b4ad368aff3a7d35b76aa8c96cb probably makes sense everywhere (and no issues reported so far, which is good).

Something else I'm observing is that this method returns `true` if an explicit branch argument is given. So I believe this should not affect Bundler, which has an explicit `:branch` option when it comes to dealing with `git` sources. In other words, I think this should only potentially affect ecosystems with an "ambiguous notation" for git refs.

Reply:
Hei @brrygrdn! Does it look ok now? I'd like to move this one forward if so.